batcher/batcher.go

```diff
@@ -158,6 +165,11 @@ func (b *basicBatcher) Dispose() {
 	b.flush()
 	b.disposed = true
 	b.items = nil
+
+	// Drain the batch channel and all routines waiting to put on the channel
+	for len(b.batchChan) > 0 || atomic.LoadInt32(&b.waiting) > 0 {
```

Comment:
Not sure if you'll ever hit it, or if it matters, but two threads could have len(b.batchChan) evaluate to greater than 0 at the same time and if there was only one the other will be blocked on the next line.